.github/scripts/init_linux_port_vm_network.sh

```diff
@@ -5,37 +5,85 @@
 #
 # This script sets up virtual interface (veth1 and veth0 ) with DHCP configure that can be used for linux port.
 
-ip link add veth0 type veth peer name veth1
-ifconfig veth0 192.168.1.1 netmask 255.255.255.0 up
-ifconfig veth1 up
-ethtool --offload veth0 tx off
-ethtool --offload veth1 tx off
-
-sh -c "cat > /etc/dhcp/dhcpd.conf" <<EOT
-subnet 192.168.1.0 netmask 255.255.255.0 {
-    range 192.168.1.100 192.168.1.200;
-    option routers 192.168.1.1;
-    option domain-name-servers 1.1.1.1;
-    option broadcast-address 192.168.1.255;
-    default-lease-time 600;
-    max-lease-time 7200;
+getNextAvailableIpRange() {
+    for (( i=1; i<=255; i++ )); do
+        match=`ifconfig -a | grep -o -E "inet $1\.$2\.$i\.[0-9]+"`
+        #echo "$match"
+
+        if [[ "$match" == "" ]]; then
+            echo $i
+            break
+        fi
+    done
 }
+
+if [[ "$1" == "--help" || "$1" == "-h" ]]; then
+    echo "Creates a virtual network interface for the freertos simulator."
+    echo "Use option --clean to undo any changes."
+elif [[ "$1" == "--clean" ]]; then
+    ifconfig veth0 down
```

Comment:
how about letting user pass interface name ? and by default we name the interface as veth0 if not passed?

how to set default value.
https://github.com/Azure-Samples/iot-middleware-freertos-samples/blob/06d8de660ca1a90e8ca8c2b03eaf4161060decfc/.github/scripts/fetch_freertos.sh#L12****

Reply:
We could... I dont' think it's a high priority, because it's unlikely the system will have a veth0 and veth1 already. I'll sync with you offline.

Reply:
Talked to Himanshu, we will not add option to set name of virtual interface, but we will rename the default names from veth0 to rtosveth0 and rtosveth1.